NGPIris/hci/hci.py

```diff
@@ -10,105 +10,74 @@
 import json
 import os
 import sys
+import urllib3
 
 from NGPIris import WD
 
-# Creates template based on template. 
-def create_template(index, query):
-    with open(f"{WD}/hci/template_query.json", "r") as sample:
-        data = json.load(sample)
-        data["indexName"] = index
-        data["queryString"] = query
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)  # Disable warnings about missing SLL certificate.
 
-    with open(f"{WD}/hci/written_query.json", "w") as dumpyboi:
-        json.dump(data, dumpyboi, indent=4)
-
-
-def generate_token(password):
-    """Generate a security token from a password"""
-    with open(password) as pw:
-        admin_pass = str(pw.readline()).strip()
-        my_key = requests.post("https://10.248.2.93:8888/auth/oauth/", data={"grant_type": "password", "username": "admin", "password": f"{admin_pass}", "scope": "*", 
-            "client_secret": "hci-client", "client_id": "hci-client", "realm": "LOCAL"}, verify=False)
-        
+class HCIManager:
+    
+    def __init__(self,credentials_path=""):
+        if credentials_path != "":
+            c = preproc.read_credentials(credentials_path)
+            self.password = c['ngpi_password']
+
+            self.address = "10.248.192.3"
+            self.authport = "8000"
+            self.apiport = "8888"
+
+    def get_password():
+        return self.password
+
+    # Creates template based on template. 
+    def create_template(self, index, query):
+        with open(f"{WD}/hci/template_query.json", "r") as sample:
+            data = json.load(sample)
+            data["indexName"] = index
+            data["queryString"] = query
+
+        with open(f"{WD}/hci/written_query.json", "w") as dumpyboi:
+            json.dump(data, dumpyboi, indent=4)
+
+
+    def generate_token(self):
+        """Generate a security token from a password"""
+        my_key = requests.post(f"https://{self.address}:{self.authport}/auth/oauth/", data={"grant_type": "password", "username": "admin", "password": f"{self.password}", "scope": "*", 
+                "client_secret": "hci-client", "client_id": "hci-client", "realm": "LOCAL"}, verify=False)
+            
         return ast.literal_eval(my_key.text)["access_token"].lstrip()
```

Comment:
This spontaneously seems like a weird way to access it but I have not seen it so I don't know...